big-red-shuttle/StopsViewController.swift

```diff
@@ -101,5 +123,186 @@ class StopsViewController: UIViewController, GMSMapViewDelegate {
         let update = GMSCameraUpdate.setTarget(CLLocationCoordinate2DMake(newLat, newLong))
         mapView.animate(with: update)
     }
+    
+    
+    /*This function displays/hides the popUpView based on tapping the marker.*/
+    func mapView(_ mapView: GMSMapView, didTap marker: GMSMarker) -> Bool {
+        if currentMarker == nil {
+            currentMarker = marker
+            popUp(stop: marker.userData as! Stop)
+        } else if currentMarker.position.latitude != marker.position.latitude {
+            dismissPopUpView(marker: marker, fullyDismissed: false)
+        } else {
+            dismissPopUpView(marker: marker, fullyDismissed: true)
+        }
+        return true
+    }
+    
+    
+    func directionsButtonPressed(sender: UIButton) {
+        //prepare to redirect user to map app
+        let location = selectedStop.getLocation()
+        if UIApplication.shared.canOpenURL(URL(string: "comgooglemaps://")!) {
```

Comment:
might be good to make the string a constant and use it below as well, so if it ever changes, you don't have to change the code in 3 places